test/mzworkflows_examples/mzworkflows.py

```diff
@@ -0,0 +1,58 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import os
+from unittest.mock import patch
+
+from materialize.mzcompose import (
+    Kafka,
+    Materialized,
+    SchemaRegistry,
+    Testdrive,
+    Workflow,
+    Zookeeper,
+)
+
+confluents = [Zookeeper(), Kafka(), SchemaRegistry()]
+
+versioned_mz = Materialized(image="materialize/materialized:${MZ_VERSION}")
+mz_instances = list(
+    map(
+        lambda i: Materialized(
+            name=f"materialized{i}",
+            data_directory=f"/share/materialized{i}",
+            port=6875 + i,
+        ),
+        [1, 2],
+    )
+)
+
+services = confluents + mz_instances + [versioned_mz, Testdrive()]
+
+
+def workflow_start_confluents(w: Workflow):
+    w.start_and_wait_for_tcp(services=confluents)
+
+
+def workflow_versioned_mz(w: Workflow):
+    for version in ["v0.7.0", "v0.8.0"]:
+        with patch.dict("os.environ", {"MZ_VERSION": version}):
```

Comment:
So, I would absolutely love if we moved to a world where mzcompose didn't deal in environment variables at all. I find them approximately impossible to reason about, since some are interpolated by mzcompose, some by Docker Compose, and some not at all. What do you think about tweaking this so that you actually generate a differently-named service for each version, like you do with `mz_instances`? "materialized-v0.7.0", "materialized-v0.8.0", and so on.

Reply:
Yes, I understand your concern. I was also thinking the same thing:  `f'{foo}'` is one thing, `${foo}` is another and `$${foo}` something else altogether.

For this problem in particular, I like your idea and I will try to come up with an env-less test as I refactor `test/upgrade/mzcompose.yml` to be in Python in a subsequent PR.

Reply:
I have fixed the example using your suggested pattern